File: articles/astronomical-tables-calendars-and-algorithmic-prediction/java/Interpolation.java
public class Interpolation {
    public static void main(String[] args) {
        double x0 = 10.0, y0 = 1.2;
        double x1 = 20.0, y1 = 2.8;
        double x = 15.0;
        double y = y0 + ((x - x0) / (x1 - x0)) * (y1 - y0);
        System.out.printf("interpolated_y=%.6f%n", y);
    }
}

File: articles/astronomical-tables-calendars-and-algorithmic-prediction/rust/interpolation.rs
fn main() {
    let x0 = 10.0_f64;
    let y0 = 1.2_f64;
    let x1 = 20.0_f64;
    let y1 = 2.8_f64;
    let x = 15.0_f64;
    let y = y0 + ((x - x0) / (x1 - x0)) * (y1 - y0);
    println!("interpolated_y={:.6}", y);
}
